packages/blade/src/components/Typography/Heading/Heading.tsx

```diff
@@ -16,11 +17,16 @@ type HeadingCommonProps = {
 
 type HeadingNormalVariant = HeadingCommonProps & {
   variant?: Exclude<HeadingVariant, 'subheading'>;
+  size?: HeadingSize;
   weight?: keyof Theme['typography']['fonts']['weight'];
 };
 
 type HeadingSubHeadingVariant = HeadingCommonProps & {
   variant?: Extract<HeadingVariant, 'subheading'>;
+  /**
+   * `size` cannot be used with variant="subheading". Either change to variant="regular" or remove size prop
+   */
+  size?: undefined;
```

Comment:
Well, this I feel like an arguably less ideal that what we had before. This introduces one more impossible state :sad:

Reply:
True but don't see any other way to do this without introducing inconsistency in prop names. This way we can atleast say that size is decided by `size` prop and subheading just becomes an exception where size can't be changed.